Fix record short syntax rewriting issue

Related to discussion in  #111 
I believe we could have a small patch that takes care of renaming records under the shortcut syntax with the missing _patched suffix.

Also, tried to add a script override, for instance adding suffix `public record LeapAction__Patched_` but FSR still rewrites it `public record LeapAction`, the constructor gets rewritten with the suffix though : `public LeapAction__Patched_()`. We end up with the compiler throwing on the ctor `error CS1520: Method must have a return type'` as the ctor is recognised as a regular function.

--- a/Assets/Scripts/Editor/Compilation/CodeRewriting/RecordRewriter.cs
+++ b/Assets/Scripts/Editor/Compilation/CodeRewriting/RecordRewriter.cs
@@ -1,0 +1,36 @@
+﻿using FastScriptReload.Runtime;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace FastScriptReload.Editor.Compilation.CodeRewriting
+{
+	class RecordeRewriter : FastScriptReloadCodeRewriterBase
+    {
+        public RecordeRewriter(bool writeRewriteReasonAsComment)
+	        : base(writeRewriteReasonAsComment)
+        {
+	        
+        }
+        
+        public override SyntaxNode VisitRecordDeclaration(RecordDeclarationSyntax node)
+        {
+	        return AdjustRecordName(node, node.Identifier);
+        }
+
+        private SyntaxNode AdjustRecordName(RecordDeclarationSyntax node, SyntaxToken nodeIdentifier)
+        {
+	        var typeName = nodeIdentifier.ToString(); //Not ToFullString() as it may include spaces and break.
+	        
+	        if (!typeName.EndsWith(AssemblyChangesLoader.ClassnamePatchedPostfix))
+	        {
+		        typeName += AssemblyChangesLoader.ClassnamePatchedPostfix;
+	        }
+
+	        return AddRewriteCommentIfNeeded(
+		        node.ReplaceToken(nodeIdentifier, SyntaxFactory.Identifier(typeName)), 
+		        $"{nameof(RecordeRewriter)}:{nameof(AdjustRecordName)}"
+		    );
+        }
+    }
+}

--- a/Assets/Scripts/Editor/Compilation/CodeRewriting/RecordRewriter.cs.meta
+++ b/Assets/Scripts/Editor/Compilation/CodeRewriting/RecordRewriter.cs.meta
@@ -1,0 +1,3 @@
+fileFormatVersion: 2
+guid: bb801c55172044ed9294ef722937f12f
+timeCreated: 1705523787

--- a/Assets/Scripts/Editor/Compilation/DynamicCompilationBase.cs
+++ b/Assets/Scripts/Editor/Compilation/DynamicCompilationBase.cs
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
@@ -184,6 +184,7 @@ namespace FastScriptReload.Editor.Compilation
                 combinedUsingStatements.AddRange(hotReloadCompliantRewriter.StrippedUsingDirectives);
                 
                 root = new BuilderPatternFunctionsRewriter(DebugWriteRewriteReasonAsComment).Visit(root);
+                root = new RecordeRewriter(DebugWriteRewriteReasonAsComment).Visit(root);
                 
                 //processed as last step to simply rewrite all changes made before
                 if (TryResolveUserDefinedOverridesRoot(tree.FilePath, definedPreprocessorSymbols, out var userDefinedOverridesRoot))

